STYLE-GUIDE.md

```diff
@@ -0,0 +1,22 @@
+# Exercism's Style Guide
+
+This document acts as a Style Guide for the language and wording used in exercises.
+
+## General principle
+
+All content should be written in US english. 
+In the future other translations may occur, but the "official" Exercism natural langauge is US English.
+
+## Explain or substitude mathmatical and esotoric terms.
+
+In any place that mathematical terms are used they should be explained or substituted out for terms that require less domain knowledge.
+
+Examples:
+- Rather than using "natural numbers", we should use "positive integers" or "zero and positive integers". 
```

Comment:
`zero`, `positive` and `integers` are mathematical terms.

Reply:
But they are more plainly understood than `natural numbers` which appears to be the intent

Reply:
I'm not convinced that "integers" is more plainly understood than "natural numbers".

Reply:
¯\\\_(ツ)_/¯

Reply:
> I'm not convinced that "integers" is more plainly understood than "natural numbers".

I'm pretty convinced it is as `Integer` is a commonly used class / type in most languages, whereas `natural` isn't a programming "concept" I've ever used.

But I'd be happy to say "positive whole numbers" too.

Reply:
Yea that's precisely the thing. I hear the term integers mostly in a programming context, while natural numbers are used everywhere and more likely to be known even by programming novices because it's a term they (probably) heard in school.

As you know, I'm really not a fan of singling out maths as some scary voodoo thing that people must be afraid of. I'd be happier if it was an example where, instead of a mathematical term, an everyday term would be used. Maybe something like `Rather than using "bijective map", we should use "1-to-1 map"`? (Not a great example either.)

> But I'd be happy to say "positive whole numbers" too.

That's worse imo because positive whole numbers are just natural numbers :D

Reply:
I'm happy to use ~zero~ no esoteric terms (where integer might be considered esoteric here). Especially as it's kinder to non-native speakers.

But, we do actively presume someone has a programming background, and we equally actively presume someone doesn't have a maths background. So we should be more comfortable using programming words than maths words.

> That's worse imo because positive whole numbers are just natural numbers :D

Sometimes natural numbers include zero, sometimes they don't :P

All concepts have words that label them. But requiring someone to know more words is a less kind thing than using two words to explain something when they might not.

Saying 
> positive whole numbers (know as "natural numbers")

Then using `natural numbers` throughout the rest of the description is fine.

Reply:
Isn't it good for understanding to do "whole numbers" (integers)" and "whole positive numbers (unsigned integers)"? If we're trying to be as _accessible_ as possible, this might be preferable.

My problem with _integer_ is that it's NOT the same thing across programming languages.

Reply:
> But, we do actively presume someone has a programming background, and we equally actively presume someone doesn't have a maths background. So we should be more comfortable using programming words than maths words.

I'd argue that high school words are an even lower requirement than programming words, therefore we should use those when possible.

That said, this is kinda pointless, maybe we can come up with a differenti example instead of arguing whether integers or natural numbers are a more commonly known term?

Reply:
```suggestion
- Rather than using "bijective map", we should use "1-to-1 map".
```

Reply:
I (genuinely) don't understand "1 to 1 map".

Wikipedia says:

> In mathematics, a bijection, bijective function, one-to-one correspondence, or invertible function, is a function between the elements of two sets, where each element of one set is paired with exactly one element of the other set, and each element of the other set is paired with exactly one element of the first set.

If we needed to use the word bijection, we would need to be using some level of information that covers this. Maybe "where each element in one set has a paired element in another set"? (I don't know if that's correct - but that's how I'm interpretting that ^^)

```suggestion
- Rather than using "natural numbers", we should use "positive whole numbers".
```

Reply:
I think if a bijection were required there would be some easy contextual way of describing the property (maybe we need things to be reversible, or maybe we just need a unique name for each robot and for there to be no names that don't correspond to robots). So this could be covered under "no jargon".

In general, I think "a one-to-one mapping" or "one-to-one correspondence" is more clear. If someone starts talking about surjections and bijections then I almost always just have to look it up. And btw, your definition is slightly wrong in that it doesn't ensure that the second set only contains elements paired with the first, nor that each element in the first set is paired with a unique element in the second.

This table is a nice way of explaining it for me:

![image](https://user-images.githubusercontent.com/6000761/96198371-8ea45e80-0f4c-11eb-8d8e-d8043fb0e228.png)